### DISCUSSION TITLE
platformio-ide-terminal

### DISCUSSION THREAD

**Discussion by Alessandro-88** [pulsar-edit/.github#231] [Community Package Help]
I have tried to install the platformio-ide-terminal package from platformio several times but it gives me an error. I have also tried to install other packages similar to this one and they give me the same error. This is the error:

Installing “platformio-ide-terminal@2.10.1” failed.Hide output…

> node-pty-prebuilt-multiarch@0.9.0 install /tmp/apm-install-dir-20231018-2944-116lnzh.d8iq/node_modules/platformio-ide-terminal/node_modules/node-pty-prebuilt-multiarch
> prebuild-install || node scripts/install.js

undefined

prebuild-install WARN install No prebuilt binaries found (target=12.2.3 runtime=electron arch=x64 libc= platform=linux)
gyp ERR! build error 
gyp ERR! stack Error: not found: make
gyp ERR! stack at getNotFoundError (/opt/Pulsar/resources/app/ppm/node_modules/npm/node_modules/node-gyp/node_modules/which/lib/index.js:16:17)
gyp ERR! stack at which (/opt/Pulsar/resources/app/ppm/node_modules/npm/node_modules/node-gyp/node_modules/which/lib/index.js:81:9)
gyp ERR! stack at async doWhich (/opt/Pulsar/resources/app/ppm/node_modules/npm/node_modules/node-gyp/lib/build.js:113:22)
gyp ERR! stack at async loadConfigGypi (/opt/Pulsar/resources/app/ppm/node_modules/npm/node_modules/node-gyp/lib/build.js:78:7)
gyp ERR! stack at async build (/opt/Pulsar/resources/app/ppm/node_modules/npm/node_modules/node-gyp/lib/build.js:36:3)
gyp ERR! System Linux 6.5.11-300.fc39.x86_64
gyp ERR! command "/opt/Pulsar/resources/app/ppm/bin/node" "/opt/Pulsar/resources/app/ppm/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /tmp/apm-install-dir-20231018-2944-116lnzh.d8iq/node_modules/platformio-ide-terminal/node_modules/node-pty-prebuilt-multiarch
gyp ERR! node -v v16.0.0
gyp ERR! node-gyp -v v9.4.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open '/tmp/apm-install-dir-20231018-2944-116lnzh.d8iq/package.json'
npm WARN apm-install-dir-20231018-2944-116lnzh.d8iq No description
npm WARN apm-install-dir-20231018-2944-116lnzh.d8iq No repository field.
npm WARN apm-install-dir-20231018-2944-116lnzh.d8iq No README data
npm WARN apm-install-dir-20231018-2944-116lnzh.d8iq No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-pty-prebuilt-multiarch@0.9.0 install: `prebuild-install || node scripts/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-pty-prebuilt-multiarch@0.9.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jaw/.pulsar/.apm/_logs/2023-11-19T06_50_12_534Z-debug.log

Help me, please.

**Comment by Daeraxa** [comment]
Out of interest do you get the same problem with `x-terminal-reloaded` which is currently maintained?

**Reply by Alessandro-88** [reply]
> Out of interest do you get the same problem with `x-terminal-reloaded` which is currently maintained?

Yes I do. I'm using fedora 39 KDE Plasma 64bit x86_64. A few days ago I was using Fedora 38 KDE plasma on the same PC and everything was ok.

**Reply by Daeraxa** [reply]
It sounds like you don't have `make` in order to build the package. You might need to install the basic Fedora build tools e.g. (`dnf install @development-tools`).It sounds like you don't have `make` in order to build the package. You might need to install the basic Fedora build tools e.g. (`dnf install @development-tools`).

**Comment by Alessandro-88** [comment]
Yes I do. I'm using fedora 39 KDE Plasma 64bit x86_64. A few days ago I was
using Fedora 38 KDE plasma on the same PC and everything was ok.

**Comment by Alessandro-88** [comment]
Thank you so much. I installed a group of C packages and libraries and was
able to install platformio-ide-terminal successfully. First I installed
x-terminal-reloaded but it didn't work well; The terminal opened to the
entire screen without letting me see the code. I looked for information and
it seems that it is a problem with node-pty, but I don't know how to
install it on Fedora.

**Reply by Daeraxa** [reply]
> The terminal opened to the entire screen without letting me see the code.

That is just an option, I have it set to open in a dock:
![image](https://github.com/pulsar-edit/.github/assets/58074586/993ba0a8-8b36-4eae-8994-84517caf18d1)

I've been meaning to bring this up as a potential improvement to the package to maybe change the default from a pane to a dock.

**Comment by Alessandro-88** [comment]
Today I installed x-terminal-reloaded on my laptop with openSUSE Tumbleweed
and it installed fine but it doesn't open the terminal in the current
directory.  Do you know how to fix this?

**Reply by Daeraxa** [reply]
Did you set the following?
![image](https://github.com/pulsar-edit/.github/assets/58074586/41e85c3f-3e68-4205-b84d-bddc7be031f2)

**Comment by Alessandro-88** [comment]
Jjj no I didn't.  I'm going to install it again and configure it
correctly.  But I think it would be better if these configurations came by
default.

**Comment by Alessandro-88** [comment]
Hi. I reinstalled x-terminal-reloaded and configured it as I pleased. It's
great. I just need to remove the blinking of the cursor. I searched and
didn't see any option to do it. Do you know how to do this?

**Reply by Whyhow314** [reply]
In the "xterm.js Terminal Options" setting, put {"cursorBlink":false}

**Comment by Alessandro-88** [comment]
You saved my life dude. Thank you very much.
Now I have a problem editing files with Neovim in this terminal.
When I press the keys strange characters appear.
Do you know how to fix this?

**Reply by Daeraxa** [reply]
I don't see anything weird at all with neovim, not that I've ever tried opening a different editor within my main editor before lol

**Reply by Alessandro-88** [reply]
Dude I just edited a file with nvim in x-terminal-reloaded and in platformio-ide-terminal. With platformio-ide-terminal everything is fine. With x-terminal-reloaded everything is wrong. I like x-terminal-reloaded better but it's making it difficult for me. In x-terminal-reloaded when using nvim when I press the keys the equivalent in integers is displayed, not the correct character. If I press 'a' it shows 'abc[+]2'

**Reply by Spiker985** [reply]
That sounds like a modifier key was seemingly pressed at the same time

In terms of the same working in platformio, it may be that something has changed inside xterm, but I can assure you outside of package bumps, and a few additional settings, it's basically unchanged from x-terminal and platformio before that

**Reply by Alessandro-88** [reply]
What is needed is to see how this problem is solved in x-terminal-reloaded.

**Reply by Daeraxa** [reply]
The first step would be to replicate it, which I haven't been able to so far.

**Reply by Alessandro-88** [reply]
replicate it? what is that?

**Reply by savetheclocktower** [reply]
Provide steps to reproduce the problem. Right now this is a problem that has happened to you, but that nobody else knows how to make happen to them. If we can't make the problem happen, we can't fix it.

**Comment by Nine-H** [comment]
Hey guys, I'm testing the windows version, I have a linux dev environment hosted in wsl2 is there a terminal plugin that can be installed and utilize wsl2 shells? I've tried `x-terminal-reloaded` and it fails to install on windows. cheers.

**Reply by Alessandro-88** [reply]
Buddy. I don't know. But I advise you to separate the Windows and GNU-Linux installations on your device. If you want to use tools that are natural to the Linux environment, use Linux and if you want to play video games, use Windows. I think it will be easier for you that way.

**Reply by Nine-H** [reply]
your post is remarkably unprofessional and condescending. the original atom did work with wsl2 shells, and I'm using windows in a professional capacity where I need to use janky, windows exclusive vendor tooling alongside development work.

**Reply by Alessandro-88** [reply]
jjj dude. Dude you have to relax. You have to have more self-esteem. Don't get offended by nonsense like that. You are a Windows and wsl2 professional, go for it.

**Reply by savetheclocktower** [reply]
@Alessandro-88 Nah, you've got to be nicer to people. It was a jerk comment; just own it, apologize, and move on.

**Reply by Alessandro-88** [reply]
jjj.  You too?  This crystal generation is offended by everything.  What's wrong with him liking to play on Windows?

**Reply by Daeraxa** [reply]
Stop. Or I just close this discussion. A technical question was asked, they didn't ask for your opinion on how they work (which you have no idea if it is enforced or not, for example I don't have a choice about using anything other than Windows at work ). If you don't know then just move on.

**Comment by Alessandro-88** [comment]
Yes you're right. I tried editing a file in x-terminal-reloaded with nano
and with nvim in openSUSE and everything worked fine. The problem is
happening to me in Fedora 39. Something happens there that the files cannot
be edited.

**Comment by Alessandro-88** [comment]
Ok. I didn't expect anyone to be offended due to someone didn't use Windows
in any other way than professionally.

**Comment by Daeraxa** [comment]
Closed and locked. Original questions was answered and resolved so if there are any further questions regarding terminal packages (tagging @Nine-H), then feel free to open a new discussion.
